src/lib/messages.ts

```diff
@@ -0,0 +1,15 @@
+import pick from "lodash/pick";
+import type { AbstractIntlMessages } from "next-intl";
+
+const CLIENT_MESSAGES = {
```

Comment:
Is it possible to make it typesafe based on `messages/en.json`?

Reply:
@AdiPol1359 just coming up with a prototype https://tsplay.dev/N7kvnN. I think that type safety here can reduce misspell significantly. What do you think?

https://github.com/user-attachments/assets/fdd9bd47-e615-4007-9ffa-6f75d44d1449

Reply:
> @AdiPol1359 just coming up with a prototype https://tsplay.dev/N7kvnN. I think that type safety here can reduce misspelling significantly. What do you think?
> 
>  demo.mov

It's a good idea, but in my opinion it doesn't make sense in this case. We need to create too complex type to correctly infer all the keys in the object. There is not much chance of making a typo, and even if it happens - it will be easy to catch it by not displaying the text correctly on the page.